File: 백준/Bronze/10818. 최소， 최대/최소， 최대.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const N = +input[0]
const arr = input[1].split(' ').map(e=>+e)

const max = arr.reduce((acc,cur)=>Math.max(acc,cur))
const min = arr.reduce((acc,cur)=>Math.min(acc,cur))

console.log(`${min} ${max}`)

File: 백준/Bronze/10818. 최소， 최대/최소， 최대.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

// https://www.acmicpc.net/problem/10818 (최소, 최대)
// 2023-08-29
public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();

    int N = Integer.parseInt(br.readLine());
    String[] input = br.readLine().split(" ");
    int[] arr = Arrays.stream(input).map(Integer::parseInt).mapToInt(i -> i).toArray();

    sb.append(Arrays.stream(arr).min().getAsInt()).append(" ")
      .append(Arrays.stream(arr).max().getAsInt());

    System.out.println(sb);
  }
}
